azblob/zt_url_blob_test.go

```diff
@@ -1970,3 +1970,192 @@ func (s *aztestsSuite) TestDownloadBlockBlobUnexpectedEOF(c *chk.C) {
+	c.Assert(err, chk.IsNil)
+	c.Assert(resp.VersionID(), chk.NotNil)
+
+	listBlobResp, err := containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Metadata: true}})
+
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobResp.Segment.BlobItems[0].Name, chk.Equals, blobName)
+	c.Assert(listBlobResp.Segment.BlobItems[0].Metadata, chk.HasLen, 2)
+	c.Assert(listBlobResp.Segment.BlobItems[0].Metadata, chk.DeepEquals, metadata)
+}
+
+func (s *aztestsSuite) TestCreateAndDownloadBlobSpecialCharactersWithVID(c *chk.C) {
+	bsu := getBSU()
+	containerURL, _ := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+	data := []rune("-._/()$=',~0123456789")
+	for i := 0; i < len(data); i++ {
+		blobName := "abc" + string(data[i])
+		blobURL := containerURL.NewBlockBlobURL(blobName)
+		resp, err := blobURL.Upload(ctx, strings.NewReader(string(data[i])), BlobHTTPHeaders{}, Metadata{}, BlobAccessConditions{})
+		c.Assert(err, chk.IsNil)
+		c.Assert(resp.VersionID(), chk.NotNil)
+
+		dResp, err := blobURL.WithVersionID(resp.VersionID()).Download(ctx, 0, CountToEnd, BlobAccessConditions{}, false)
+		d1, err := ioutil.ReadAll(dResp.Body(RetryReaderOptions{}))
+		c.Assert(err, chk.IsNil)
+		c.Assert(dResp.Version(), chk.Not(chk.Equals), "")
+		c.Assert(string(d1), chk.DeepEquals, string(data[i]))
+		versionId := dResp.r.rawResponse.Header.Get("x-ms-version-id")
+		c.Assert(versionId, chk.NotNil)
+		c.Assert(versionId, chk.Equals, resp.VersionID())
+	}
+}
+
+func (s *aztestsSuite) TestDeleteSpecificBlobVersion(c *chk.C) {
+	bsu := getBSU()
+	containerURL, _ := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+	blobURL, _ := getBlockBlobURL(c, containerURL)
+
+	blockBlobUploadResp, err := blobURL.Upload(ctx, bytes.NewReader([]byte("data")), BlobHTTPHeaders{},
+		basicMetadata, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(blockBlobUploadResp.VersionID(), chk.NotNil)
+	versionId1 := blockBlobUploadResp.VersionID()
+
+	blockBlobUploadResp, err = blobURL.Upload(ctx, bytes.NewReader([]byte("updated_data")), BlobHTTPHeaders{},
+		basicMetadata, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(blockBlobUploadResp.VersionID(), chk.NotNil)
+
+	listBlobsResp, err := containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Versions: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobsResp.Segment.BlobItems, chk.HasLen, 2)
+
+	// Deleting previous version snapshot.
+	deleteResp, err := blobURL.WithVersionID(versionId1).Delete(ctx, DeleteSnapshotsOptionNone, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(deleteResp.StatusCode(), chk.Equals, 202)
+
+	listBlobsResp, err = containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Versions: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobsResp.Segment.BlobItems, chk.NotNil)
+	if len(listBlobsResp.Segment.BlobItems) == 0 {
```

Comment:
>len(listBlobsResp.Segment.BlobItems) == 0 [](start = 4, length = 41)

can we be more precise with the validation? Instead of just making sure it's not 0, make sure that it's actually 1

Reply:
Done.